pkg/ddl/backfilling_operators.go

```diff
@@ -112,6 +123,30 @@ func (ctx *OperatorCtx) OperatorErr() error {
 	return *err
 }
 
+var (
+	_ RowCountListener = (*EmptyRowCntListener)(nil)
+	_ RowCountListener = (*distTaskRowCntListener)(nil)
+	_ RowCountListener = (*localRowCntListener)(nil)
+)
+
+// RowCountListener is invoked when some index records are flushed to disk or imported to TiKV.
+type RowCountListener interface {
+	Written(rowCnt int)
+	SetTotal(total int)
+}
+
+// EmptyRowCntListener implements a noop RowCountListener.
+type EmptyRowCntListener struct {
+}
```

Comment:
```suggestion
type EmptyRowCntListener struct {}
```

Also for below two functions, just a matter of preference